src/services/io/useIO.ts

```diff
@@ -442,71 +447,121 @@ export const useIO = create<IOStore>((set, get) => ({
+    try {
+      const timeline: TimelineStore = useTimeline.getState()
+      const { clap, segments: timelineSegments } = timeline
+      const segments: ExportableSegment[] = timelineSegments
+        .map((segment, i) => formatSegmentForExport(segment, i))
+        .filter(({ isExportableToFile }) => isExportableToFile)
+
+      let files: fflate.AsyncZippable = {}
+
+      files['screenplay.txt'] = fflate.strToU8(clap.meta.screenplay)
+
+      files['meta.json'] = fflate.strToU8(JSON.stringify(clap.meta, null, 2))
+
+      const shotcutMltXml = await generateMLT()
+      files['shotcut_project.mlt'] = fflate.strToU8(shotcutMltXml)
+
+      const videos: FFMPegVideoInput[] = []
+      const audios: FFMPegAudioInput[] = []
+
+      segments.forEach(
+        ({
+          segment,
+          prefix,
+          filePath,
+          assetUrl,
+          assetSourceType,
+          isExportableToFile,
+        }) => {
+          // we extract the base64 files
+          if (isExportableToFile) {
+            files[filePath] = [
+              // we don't compress assets since normally they already use
+              // some form of compression (PNG, JPEG, MP3, MP4 etc..)
+              base64DataUriToUint8Array(segment.assetUrl),
+              { level: 0 },
+            ]
+            assetUrl = filePath
+            assetSourceType = ClapAssetSource.PATH
+
+            if (filePath.startsWith('video/')) {
+              videos.push({
+                data: base64DataUriToUint8Array(segment.assetUrl),
+                startTimeInMs: segment.startTimeInMs,
+                endTimeInMs: segment.endTimeInMs,
+                durationInSecs: segment.assetDurationInMs / 1000,
+              })
+            }
+
+            if (
+              filePath.startsWith('music/') ||
+              filePath.startsWith('dialogue/')
+            ) {
+              audios.push({
+                data: base64DataUriToUint8Array(segment.assetUrl),
+                startTimeInMs: segment.startTimeInMs,
+                endTimeInMs: segment.endTimeInMs,
+                durationInSecs: segment.assetDurationInMs / 1000,
+              })
+            }
+          }
+
+          // segment metadata
+          files[`segments/${prefix}${segment.id}.json`] = fflate.strToU8(
+            JSON.stringify(
+              {
+                ...segment,
+                assetUrl,
+                assetSourceType,
+              },
+              null,
+              2
+            )
+          )
         }
+      )
 
-        // segment metadata
-        files[`segments/${prefix}${segment.id}.json`] = fflate.strToU8(
-          JSON.stringify(
-            {
-              ...segment,
-              assetUrl,
-              assetSourceType,
-            },
-            null,
-            2
-          )
-        )
-      }
-    )
+      const fullVideo = await createFullVideo(
+        videos,
+        audios,
+        1024,
+        576,
```

Comment:
here I think we should use:

```suggestion
        clap.meta.width,
        clap.meta.height,
```

(normally the clap format guarantees there are valid numeric values in here, as they are parsed and with min/max/defaults values)